From: Andrii Nakryiko <andrii@kernel.org>
Subject: fs/procfs: improve PROCMAP_QUERY's compat mode handling
Date: Mon, 1 Jul 2024 10:48:05 -0700

Kernel provides compat_ptr_ioctl() wrapper to sanitize pointers for 32-bit
processes on 64-bit host architectures.  Given procfs_procmap_ioctl()
always expects pointer argument, this is exactly what we need.

This has any effect only on 32-bit processes on s390 architecture.

Link: https://lkml.kernel.org/r/20240701174805.1897344-2-andrii@kernel.org
Fixes: 3757be498749 ("fs/procfs: implement efficient VMA querying API for /proc/<pid>/maps")
Signed-off-by: Andrii Nakryiko <andrii@kernel.org>
Suggested-by: Arnd Bergmann <arnd@arndb.de>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 fs/proc/task_mmu.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/fs/proc/task_mmu.c~fs-procfs-implement-efficient-vma-querying-api-for-proc-pid-maps-fix
+++ a/fs/proc/task_mmu.c
@@ -614,7 +614,7 @@ const struct file_operations proc_pid_ma
 	.llseek		= seq_lseek,
 	.release	= proc_map_release,
 	.unlocked_ioctl = procfs_procmap_ioctl,
-	.compat_ioctl	= procfs_procmap_ioctl,
+	.compat_ioctl	= compat_ptr_ioctl,
 };
 
 /*
_
